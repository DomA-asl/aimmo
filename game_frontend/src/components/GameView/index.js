import styled from 'styled-components'
import React, { Component } from 'react'
import Unity from "react-unity-webgl"

const GameViewLayout = styled.div`
  background-color: purple
  grid-area: game-view
`

export class GameView extends Component {
  render() {
    return (
      <GameViewLayout>
        <Unity
          src="TODO"
          loader="TODO"
        />
      </GameViewLayout>
    )
  }
}
export default GameView
