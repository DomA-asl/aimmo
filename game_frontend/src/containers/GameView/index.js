import styled from 'styled-components'
import React, { Component } from 'react'
import Unity, { RegisterExternalListener, UnityEvent } from 'react-unity-webgl'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { actions } from 'features/Editor'

const GameViewLayout = styled.div`
  background-color: purple
  grid-area: game-view
`

export class GameView extends Component {
  constructor(props) {
    super(props)

    this.props.getConnectionParams()

    RegisterExternalListener("SendAllConnect", this.sendAllConnect.bind(this))
  }

  sendAllConnect() {
    this.props.emitUnityEvent("World Controller", "SetGameURL", this.props.gameURL)
    this.props.emitUnityEvent("World Controller", "SetGamePort", this.props.gamePort)
    this.props.emitUnityEvent("World Controller", "SetGamePath", this.props.gamePath)
    this.props.emitUnityEvent("World Controller", "SetSSL", this.serialisedSSLFlag())
    this.props.emitUnityEvent("World Controller", "EstablishConnection", "k")
  }

  serialisedSSLFlag() {
    let boolString = this.props.sslFlag.toString()
    
    return boolString.charAt(0).toUpperCase() + boolString.slice(1)
  }
  
  render() {
    return (
      <GameViewLayout>
        <Unity
          src="/static/unity/Build/unity.json"
          loader="/static/unity/Build/UnityLoader.js"
        />
      </GameViewLayout>
    )
  }
}

GameView.propTypes = {
  gameURL: PropTypes.string,
  gamePath: PropTypes.string,
  gamePort: PropTypes.number,
  sslFlag: PropTypes.bool,
  getConnectionParams: PropTypes.func
}

const mapStateToProps = state => ({
  // TODO: issue #619
  gameURL: state.editor.connectionParams.game_url_base,
  gamePath: state.editor.connectionParams.game_url_path,
  gamePort: state.editor.connectionParams.game_url_port,
  sslFlag: state.editor.connectionParams.game_ssl_flag
})

const mapDispatchToProps = {
  getConnectionParams: actions.getConnectionParamsRequest,
  emitUnityEvent: actions.emitUnityEvent
}

export default connect(mapStateToProps, mapDispatchToProps)(GameView)
