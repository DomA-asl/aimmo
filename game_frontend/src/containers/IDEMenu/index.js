import styled from 'styled-components'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { actions } from 'features/Editor'
import submit from './run-code.svg'

const IDEMenuLayout = styled.nav`
  background-color: pink
  grid-area: ide-menu
`
const PostCodeButton = styled.input`
  float: right
  position: relative
  top: 18%
  right: 8%
`

export class IDEMenu extends Component {
  render () {
    return (
      <IDEMenuLayout>
        <PostCodeButton type='image' src={submit} id='post-code-button' onClick={this.props.postCode} />
      </IDEMenuLayout>
    )
  }
}

IDEMenu.propTypes = {
  postCode: PropTypes.func
}

const mapStateToProps = () => ({})

const mapDispatchToProps = {
  postCode: actions.postCodeRequest
}

export default connect(mapStateToProps, mapDispatchToProps)(IDEMenu)
