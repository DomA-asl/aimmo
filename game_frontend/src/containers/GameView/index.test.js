/* eslint-env jest */
import React from 'react'
import { GameView } from 'containers/GameView'
import { shallow } from 'enzyme/build/index'

describe('<GameView />', () => {
  it('matches snapshot', () => {

    const props = {
      gameURL: 'test',
      gamePath: '/test',
      gamePort: 8000,
      gameSSL: false,
      getConnectionParams: jest.fn(),
      setGameURL: jest.fn(),
      setGamePath: jest.fn(),
      setGamePort: jest.fn(),
      setGameSSL: jest.fn(),
      establishGameConnection: jest.fn()
    }

    const component = shallow(<GameView {...props} />)
    expect(component).toMatchSnapshot()
  })
})
