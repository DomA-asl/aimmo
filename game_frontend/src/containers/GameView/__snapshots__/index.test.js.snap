// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameView /> matches snapshot 1`] = `
<styled.div>
  <Unity
    loader="/static/unity/Build/UnityLoader.js"
    src="/static/unity/Build/unity.json"
  />
</styled.div>
`;
